### DISCUSSION TITLE
task group

### DISCUSSION THREAD

**Discussion by Antalares** [PrefectHQ/prefect#4279] [Q&A]
Hi,

I test prefect as replacement our actual scheduler (dollar universe)

We have DWH loading workflow with more than 500 steps. I try to reproduce this load with prefect but i'm facing an usability problem.
Actually our workflow steps are group by functional purpose for readability.

It seems that prefect don't have grouping facility like it's exist in airflow 2.0 (taskgroup)

Our batch workload run on daily basis schedule and have many dependencies. We consider as success pour daily loading when the complete workflow is done.
Due to the number of tasks without task group functionality it is impossible to have a readable workflow.
I look to startflow task but this put in concurrency problems.

Is there any plan to implement task group functionality for readability purpose ?

Thanks you for your answers.
Alex

**Comment by Antalares** [comment]
Below the task group implentation in airflow :
https://cwiki.apache.org/confluence/plugins/servlet/mobile?contentId=148645525#content/view/148645525

It's exactely I want to solve :
    - Makes it possible to organize DAGs with large number tasks into more manageable groups.
    - Makes UI easier to use especially for DAGs with many tasks

**Comment by Antalares** [comment]
Hi,

Any idea or improvement planned ?

Thank you

**Reply by zanieb** [reply]
Hey @Antalares -- we  have task 'tags' for this feature, https://docs.prefect.io/core/concepts/tasks.html#tags

**Reply by zanieb** [reply]
I do see that the 'Task Group' collapses to a single task in the UI -- this is something we're interested in adding but it hasn't made it on our roadmap yet.

**Reply by LINKDHS-ITS** [reply]
Hey @Antalares, I had a similar problem and I've successfully tested (though not in much depth) putting groups of tasks into a normal python function. 
    
    @task
    def test2():
        logger.info('Run the second task')

    @task
    def test1():
        logger.info('Run the first task')

    def group_func():
        test1()
        test2()
    

    with Flow('Group Tasks 1') as flow1:
        group_func()

    with Flow('Group Tasks 2') as flow2:
        group_func()
    
    flow1.run()
    flow2.run()
    
You will not get the benefits of the group being a task itself (like state), but it has the benefit of being a function that can be called by multiple flows or imported from another file which should help with readability and organization. However, I haven't tested this fully in a production environment so I do not know what issues might surface from doing this.
